packages/purpose-process/src/services/validators.ts

```diff
@@ -295,10 +297,8 @@ export const assertRequesterIsProducer = async ({
   producerId: TenantId;
   readModelService: ReadModelService;
 }): Promise<void> => {
-  const activeProducerDelegation = await readModelService.getActiveDelegation(
-    eserviceId,
-    delegationKind.delegatedProducer
-  );
+  const activeProducerDelegation =
```

Comment:
To align this with the agreement process and with the consumer assertion below I would:
-  rename this function to `assertRequesterCanActAsProducer`
- do the same as the `assertRequesterCanActAsConsumer`, having the two separate assertions called in the if (assertRequesterIsProducer and assertRequesterIsDelegateProducer)
-  move the retrieval of the delegation in the service and outside this function